cmd/build/v2/build.go

```diff
@@ -341,13 +343,24 @@ func (bc *OktetoBuilder) buildSvcFromDockerfile(ctx context.Context, manifest *m
 	bc.ioCtrl.Logger().Info(fmt.Sprintf("Building service '%s' from Dockerfile", svcName))
 	isStackManifest := manifest.Type == model.StackType
 	buildSvcInfo := bc.getBuildInfoWithoutVolumeMounts(manifest.Build[svcName], isStackManifest)
-	var err error
 	var buildHash string
 	if bc.smartBuildCtrl.IsEnabled() {
 		buildHash = bc.smartBuildCtrl.GetBuildHash(buildSvcInfo, svcName)
 	}
-	tagToBuild := newImageTagger(bc.Config, bc.smartBuildCtrl).getServiceImageReference(manifest.Name, svcName, buildSvcInfo, buildHash)
-	buildSvcInfo.Image = tagToBuild
+	it := newImageTagger(bc.Config, bc.smartBuildCtrl)
+	tagsToBuild := it.getServiceImageReference(manifest.Name, svcName, buildSvcInfo, buildHash)
+	imageCtrl := registry.NewImageCtrl(bc.oktetoContext)
+	globalImage := it.getGlobalTagFromDevIfNeccesary(tagsToBuild, bc.oktetoContext.GetNamespace(), bc.oktetoContext.GetRegistryURL(), buildHash, manifest.Name, svcName, imageCtrl)
+	if globalImage != "" {
+		tagsToBuild = fmt.Sprintf("%s,%s", tagsToBuild, globalImage)
+	}
+	if buildSvcInfo.Image == "" {
```

Comment:
How hard would be to generate always the dev tag when image is empty, so we don't need to account for this special case?

I mean, this has an implicit knowledge. When the image is not specified, we are building the image for the global registry (don't know exactly why we build that one), and then, we generate the dev one. It is hard to remember to change this if we change which image is built first in that case.

There shouldn't be special cases and consider everything the same as much as possible for simplicity and code maintainability.

If we don't change it, at least, let's add a comment before the if to specify why we need this if